src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -209,10 +273,23 @@ public Stream<String> getBlobsLinesStream(
   @Override
   public void validateUserCanRead(
       List<String> sourcePaths, String cloudEncapsulationId, AuthenticatedUserRequest user) {
-    // This checked is not needed for Azure because we use signed URLS that by default check
-    // permissions
-    // Keeping this method because we do need to do this check for GCP in code that is shared
-    // between gcp and azure [See CloudFileReader]
+    // This check is not needed for Azure source because we use signed URLS that by default check
+    // permissions but this is needed if ingesting from a GCS hosted file
+    List<String> gsPaths = sourcePaths.stream().filter(p -> p.startsWith("gs://")).toList();
+
+    // Extract the project ids from the userProject query parameter.  There can be 0 or 1 values
+    List<String> cloudEncapsulationIds =
+        gsPaths.stream()
+            .map(p -> UriUtils.getValueFromQueryParameter(p, USER_PROJECT_QUERY_PARAM))
+            .filter(Objects::nonNull)
+            .distinct()
+            .toList();
+    if (cloudEncapsulationIds.size() > 1) {
+      throw new IllegalArgumentException("Only a single billing project per ingest may be used");
+    }
+
+    gcsPdao.validateUserCanRead(
+        gsPaths, cloudEncapsulationIds.stream().findFirst().orElse(null), user, false);
```

Comment:
I think you want `orElseThrow()` here since `cloudEncapsulationId` can't be null (can it?).
```suggestion
        gsPaths, cloudEncapsulationIds.stream().findFirst().orElseThrow(), user, false);
```

Also maybe change `size() > ` to `size() == 1` to preflight this check. Then you can replace the `stream` with `cloudEncapsulationIds.get(0)`.

Reply:
It can actually be null.  That means no project to bill to (which for loading into Azure is good)

Reply:
OK. It will result in a key entry that contains `null` since it's used in `validateUserCanRead()` like this:
```
String.format("%s%s%s", user.getToken(), PSA_SEPARATOR, cloudEncapsulationId)
```

Reply:
Ah yeah, great point.  I suppose you can't name a project null so it's moot but I agree, that's gross.  I'll fix it